weblate/api/views.py

```diff
@@ -951,6 +952,23 @@ def delete_links(self, request, project__slug, slug, project_slug):
         instance.links.remove(project)
         return Response(status=HTTP_204_NO_CONTENT)
 
+    @action(detail=True, methods=["get"], url_path="translations/zip")
+    def download_archive(self, request, **kwargs):
+        # Implementation is analogous to files#download_component, but we can't reuse
+        #  that here because the lookup for the component is different
+        instance = self.get_object()
+        if not request.user.has_perm("translation.download", instance):
+            self.permission_denied(
+                request, "Can not download all translations for the component"
+            )
+
+        instance.commit_pending("download", None)
+        return download_multi(
+            instance.translation_set.all(),
```

Comment:
I've changed the API in a6e9883101 to cover committing as well (because the commit logic is more complex since dd5a3293a97f9cf328b0b3d62030f7011f95a8d2):

```suggestion
        return download_multi(
            instance.translation_set.all(),
            [instance],
```